targets/unity-v2/Testing/Tests/Entity/EntityApiTests.cs

```diff
@@ -141,6 +145,194 @@ private void GetObjectCallback2(GetObjectsResponse result)
+                {
+                    Id = result.EntityToken.Entity.Id,
+                    Type = entityType,
+                    TypeString = result.EntityToken.Entity.TypeString
+                };
+
+                LoadFiles(testContext);
+            }
+            else
+            {
+                testContext.EndTest(UUnitFinishState.FAILED, "Entity Token is null!");
+            }
+        }
+        private void LoadFiles(UUnitTestContext testContext)
+        {
+            var request = new GetFilesRequest
+            {
+                Entity = _entityKey
+            };
+
+            PlayFabEntityAPI.GetFiles(request, PlayFabUUnitUtils.ApiActionWrapper<EntityModels.GetFilesResponse>(testContext, OnGetFilesInfo), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+        }
+        void OnGetFilesInfo(GetFilesResponse result)
+        {
+            var testContext = (UUnitTestContext)result.CustomData;
+            bool testFileFound = false;
+            GetFileMetadata fileMetaData = new GetFileMetadata();
+
+            foreach (var eachFilePair in result.Metadata)
+            {
+                if (eachFilePair.Key.Equals(TEST_FILE_NAME))
+                {
+                    testFileFound = true;
+
+                    fileMetaData = eachFilePair.Value;
+                    break; // this test only support one file
+                }
+            }
+
+            if (!testFileFound)
+            {
+                UploadFile(TEST_FILE_NAME, testContext);
+            }
+            else
+            {
+                GetActualFile(fileMetaData, testContext);
+            }
+        }
+        void GetActualFile(GetFileMetadata fileData, UUnitTestContext testContext)
+        {
+            PlayFabHttp.SimpleGetCall(fileData.DownloadUrl,
+                result =>
+                {
+                    _testFileData = Encoding.UTF8.GetString(result);
+                    TestFileContent(testContext);
+                },
+                error =>
+                {
+                    testContext.EndTest(UUnitFinishState.FAILED, error);
+                });
+        }
+        void UploadFile(string fileName, UUnitTestContext testContext)
+        {
+            var request = new InitiateFileUploadsRequest
+            {
+                Entity = _entityKey,
+                FileNames = new List<string>
+                {
+                    fileName
+                },
+            };
+
+            _testContext = testContext;
+            PlayFabEntityAPI.InitiateFileUploads(request, OnInitFileUpload, OnInitFailed);
+        }
+        void DeleteFiles(List<string> fileName, UUnitTestContext testContext)
+        {
+            var request = new DeleteFilesRequest
+            {
+                Entity = _entityKey,
+                FileNames = fileName,
+            };
+
+            PlayFabEntityAPI.DeleteFiles(request, result =>
+            {
```

Comment:
Deletefiles success callback has no contents so if it gets here, the test will time out and fail.

Reply:
Callback(lambda) here is empty, but code is correct. This will not fail, it will just continue without any additional actions.

Reply:
Perhaps I'm confused about the code flow (callbacks can get messy no matter what).
How does this get to DeleteFiles, and what happens next if it's called.